uniq/src/main.rs

```diff
@@ -0,0 +1,291 @@
+use std::{
+    fs::File,
+    io::{self, BufRead, BufReader, ErrorKind, Read, Write},
+    process,
+};
+
+mod cli;
+
+
+fn main() {
+    let matches = cli::create_app().get_matches();
+    let input_filename = matches.value_of("INPUT").unwrap_or("-");
+    let output_filename = matches.value_of("OUTPUT").unwrap_or("-");
+    let flags = Flags::from_matches(&matches);
+
+    let reader: Box<dyn io::Read> = if input_filename == "-" {
+        Box::new(io::stdin())
+    } else {
+        Box::new(File::open(input_filename).unwrap_or_else(|err| {
+            eprintln!("uniq: Cannot open '{}' for reading: {}.", input_filename, err);
+            process::exit(1);
+        }))
+    };
+    let mut reader = BufReader::new(reader);
+
+    let mut writer: Box<dyn io::Write> = if output_filename == "-" {
+        Box::new(io::stdout())
+    } else {
+        Box::new(File::create(output_filename).unwrap_or_else(|err| {
+            eprintln!("uniq: Cannot create '{}' for writing: {}.", output_filename, err);
+            process::exit(1);
+        }))
+    };
+
+    if let Err(_) = uniq(&mut reader, &mut writer, flags) {
+        process::exit(1);
+    };
+}
+
+
+#[derive(Default)]
+struct Flags {
+    show_count:       bool,        // -c | --show_count
+    supress_unique:   bool,        // -d | --repeated
+    supress_repeated: bool,        // -u | --unique
+    skip_chars:       Option<u64>, // -s | --skip-chars=N
+    skip_fields:      Option<u64>, // -f | --skip-fields=N
+}
+
+impl Flags {
+    fn from_matches(matches: &clap::ArgMatches) -> Self {
+        // Used to capture skip_chars and skip_fields
+        let parse_arg_to_u64 = |arg: Option<&str>, error_msg: &str| -> Option<u64> {
```

Comment:
I think you don't need to annotate every type in this closure, maybe just arg?

Reply:
To change this, I might as well change the name of closure variable, because it suggests that the return type is `u64`.

I'll replace it with `try_parse_arg_to_u64`.

Avoiding implicitness.

Reply:
Or maybe `parse_arg_to_option_u64`?.